### DISCUSSION TITLE
I am receiving twice the atribute_updated

### DISCUSSION THREAD

**Discussion by markicos** [zigpy/zigpy#698] [Q&A] [answered]
Hello,

I have configured the reporting for the atribute, and the listener, the problem is that i see double updated

INFO:__main__:Received an attribute update 0=0 on cluster 
INFO:__main__:Received an attribute update 0=0 on cluster 
INFO:__main__:Received an attribute update 1=1 on cluster 
INFO:__main__:Received an attribute update 1=1 on cluster 
INFO:__main__:Received an attribute update 2=2 on cluster 
INFO:__main__:Received an attribute update 2=2 on cluster 
INFO:__main__:Received an attribute update 3=3 on cluster 
INFO:__main__:Received an attribute update 3=3 on cluster 
INFO:__main__:Received an attribute update 4=4 on cluster 
INFO:__main__:Received an attribute update 4=4 on cluster 
INFO:__main__:Received an attribute update 5=5 on cluster 
INFO:__main__:Received an attribute update 5=5 on cluster 
INFO:__main__:Received an attribute update 6=6 on cluster <
INFO:__main__:Received an attribute update 6=6 on cluster <
INFO:__main__:Received an attribute update 7=7 on cluster <
INFO:__main__:Received an attribute update 7=7 on cluster <
INFO:__main__:Received an attribute update 8=8 on cluster <
INFO:__main__:Received an attribute update 8=8 on cluster <
INFO:__main__:Received an attribute update 9=9 on cluster <
INFO:__main__:Received an attribute update 9=9 on cluster <
INFO:__main__:Received an attribute update 10=10 on cluster
INFO:__main__:Received an attribute update 10=10 on cluster
INFO:__main__:Received an attribute update 11=11 on cluster
INFO:__main__:Received an attribute update 11=11 on cluster
INFO:__main__:Received an attribute update 12=12 on cluster
INFO:__main__:Received an attribute update 12=12 on cluster
INFO:__main__:Received an attribute update 13=13 on cluster
INFO:__main__:Received an attribute update 13=13 on cluster
INFO:__main__:Received an attribute update 14=5 on cluster 
INFO:__main__:Received an attribute update 14=5 on cluster 
INFO:__main__:Received an attribute update 15=15 on cluster
INFO:__main__:Received an attribute update 15=15 on cluster

**Comment by MattWestb** [comment, marked as answer]
One question wot device is you configuring the attribute reporting ?

And need some more deltas logs for see wot the device is sending to Zigpy and wot Zigpy / ZHA is requesting from the device.

If you is using one no ZCL cluster (like manufacture clusters) you need sending one default response to the device after receiving one frame from the cluster (the Zigbee stack is ding it for unicast ZCL frames).

**Reply by MattWestb** [reply]
I think you is having the problem wit https://github.com/zigpy/zigpy/discussions/689 and your device is sending one manufacture command from one no ZCL cluster (manufacture cluster) and then you need sending one default response to the device so is knowing you have revived the frame or the device is reseeding the command and its also can being that is getting problem with the device if you dont do that. 

Manufacture commands is out of ZCL that Zigbee stack is manage (that is doing the acking off all ZCL manager commands to / from ZCL clusters) but you is doing App <=> App communication and you must self manager the acking of commands sensed / received.

**Reply by markicos** [reply]
I wrote the #689 last week, now as I said the device is not reporting twice  ( I can see just one frame with an sniffer), the problem is the cluster having 2  listeners (so that I receive several attribute update) as the call back  `device_initialized` is called several times the sentence ` cluster.add_context_listener(self)` is adding several listeners. 

I think I have to figure out how to avoid initialize a device already initialized. 



```
def device_initialized(self, device, *, new=True):
        """
        Called at runtime after a device's information has been queried.
        I also call it on startup to load existing devices from the DB.
        """
        LOGGER.info("Device is ready: new=%s, device=%s", new, device)

        for ep_id, endpoint in device.endpoints.items():
            # Ignore ZDO
            if ep_id == 0:
                continue

            # You need to attach a listener to every cluster to receive events
            for cluster in endpoint.in_clusters.values():
                # The context listener passes its own object as the first argument
                # to the callback
                cluster.add_context_listener(self)
```
